velocity_modelling/write/hdf5.py

```diff
@@ -0,0 +1,283 @@
+        </DataItem>
+      </Attribute>
+      <Attribute Name="S-wave Velocity" AttributeType="Scalar" Center="Node">
+        <DataItem Dimensions="{nz} {ny} {nx}" NumberType="Float" Precision="4" Format="HDF">
+          {hdf5_relative}:/properties/vs
+        </DataItem>
+      </Attribute>
+      <Attribute Name="Density" AttributeType="Scalar" Center="Node">
+        <DataItem Dimensions="{nz} {ny} {nx}" NumberType="Float" Precision="4" Format="HDF">
+          {hdf5_relative}:/properties/rho
+        </DataItem>
+      </Attribute>
+      <Attribute Name="Basin Membership" AttributeType="Scalar" Center="Node">
+        <DataItem Dimensions="{nz} {ny} {nx}" NumberType="Int" Precision="1" Format="HDF">
+          {hdf5_relative}:/properties/inbasin
+        </DataItem>
+      </Attribute>
+    </Grid>
+  </Domain>c
+</Xdmf>
+"""
+
+    try:
+        with open(xdmf_file, "w") as f:
+            f.write(xdmf_content)
+        logger.log(logging.INFO, f"Created ParaView-compatible XDMF file: {xdmf_file}")
+    except Exception as e:
+        if isinstance(e, (SystemExit, KeyboardInterrupt)):
+            raise  # Re-raise critical exceptions
+        logger.log(logging.ERROR, f"Error creating XDMF file: {e}")
+        raise RuntimeError(f"Failed to create XDMF file: {e}")
+
+
+def write_global_qualities(
+    out_dir: Path,
+    partial_global_mesh: PartialGlobalMesh,
+    partial_global_qualities: PartialGlobalQualities,
+    lat_ind: int,
+    vm_params: dict,
+    logger: Optional[Logger] = None,
+) -> None:
+    """
+    Write a latitude slice of velocity data to a single consolidated HDF5 file.
+
+    For the first slice (lat_ind=0), this creates the file and initializes the structure.
+    For subsequent slices, it adds data to the existing file.
+
+    Parameters
+    ----------
+    out_dir : Path
+        Path to the output directory where the HDF5 file will be written.
+    partial_global_mesh : PartialGlobalMesh
+        Mesh data for the current latitude slice.
+    partial_global_qualities : PartialGlobalQualities
+        Velocity and density data for the current latitude slice.
+    lat_ind : int
+        Latitude index to determine the write mode (write or append).
+    vm_params : dict
+        Dictionary containing velocity model parameters from nzcvm.cfg.
+    logger : Logger, optional
+        Logger instance for logging messages.
+
+    Raises
+    ------
+    OSError
+        If the HDF5 file cannot be written due to permissions or disk issues.
+    """
+    # Output filename - single file for all slices
+    hdf5_file = Path(out_dir) / "velocity_model.h5"
+
+    # Apply minimum VS constraint
+    min_vs = vm_params.get(
+        "min_vs", 0.0
+    )  # Get the minimum Vs value from the parameters
+    vs_data = np.copy(partial_global_qualities.vs)
+    vs_data[vs_data < min_vs] = min_vs
+
+    ny = vm_params.get(
+        "ny"
+    )  # This is a slice along y (=lat) axis, so partial_global_mesh has only one y value
```

Comment:
When `ny` is `None` how is this handled? Currently it looks like you don't handle the `None` case explicitly and just pass it directly through to the rest of the code. Either provide a default value or behaviour for the `ny = None` case or revert to `['ny']` to throw an error right away. 
```suggestion
    ny = vm_params['ny'] # This is a slice along y (=lat) axis, so partial_global_mesh has only one y value
```

Reply:
Good suggestion. 
While this line (https://github.com/ucgmsim/velocity_modelling/blob/ce729d8ad47e91475c74c906703ece4ae0ad975e/velocity_modelling/scripts/nzcvm.py#L515) should make `vm_params['ny'] ` available, it pays to be defensive.